From 55ac9f7a9db3857ff163362855472b24fa2ba077 Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Mon, 9 Mar 2026 15:31:48 +0100
Subject: [PATCH] ibm: system1: Add CPU temperature sensors to PID loop

Ensure that the CPUs don't overheat by adding them to the PID loop.

Upstream-Status: Pending
Change-Id: Ic4c4c10ab3fea5cd53c2f58805a1134b0d72a42f
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 configurations/ibm/system1_baseboard.json | 27 +++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/configurations/ibm/system1_baseboard.json b/configurations/ibm/system1_baseboard.json
index 4a91bf2..ed51ba4 100644
--- a/configurations/ibm/system1_baseboard.json
+++ b/configurations/ibm/system1_baseboard.json
@@ -592,6 +592,33 @@
                 "CECIO"
             ]
         },
+        {
+            "Class": "temp",
+            "FFGainCoefficient": 0,
+            "FFOffCoefficient": 0,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
+            "ILimitMin": 2500,
+            "InputUnavailableAsFailed": false,
+            "Inputs": [
+                "Die_CPU0",
+                "Die_CPU1"
+            ],
+            "Name": "CPU Temperature",
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
+            "OutLimitMin": 2500,
+            "Outputs": [],
+            "PCoefficient": -500,
+            "PositiveHysteresis": 0,
+            "SetPoint": 85,
+            "SlewNeg": 0,
+            "SlewPos": 0,
+            "Type": "Pid",
+            "Zones": [
+                "CECIO"
+            ]
+        },
         {
             "Address": "0x4c",
             "Bus": 70,
-- 
2.53.0

